src/Bicep.Core.UnitTests/Diagnostics/BicepConfigSchemaTests.cs

```diff
@@ -126,5 +123,35 @@ public void RuleConfigs_RuleDescriptionsShouldBeConsistent()
                 description.Should().NotContain("\"", "use single quotes instead of double quotes in rule descriptions");
             }
         }
+
+        [TestMethod]
+        public void NoHardCodedEnvUrls_DefaultsShouldMatchInConfigAndSchema()
+        {
+            const string configPath = "src/Bicep.Core/Configuration/bicepconfig.json";
+            const string schemaPath = "src/vscode-bicep/schemas/bicepconfig.schema.json";
+
+            // From schema
+            var (rules, schema) = GetRulesAndSchema();
+            IDictionary<string, JObject>? ruleConfigs = schema.SelectToken("properties.analyzers.properties.core.properties.rules.properties")!.ToObject<IDictionary<string, JObject>>();
+            Assert.IsNotNull(ruleConfigs);
+            string[] disallowedHostsInSchema = ruleConfigs["no-hardcoded-env-urls"].SelectToken("allOf[0].properties.disallowedhosts.default")!.Values().Select(v => v.ToString()).ToArray();
+            string[] excludedHostsInSchema = ruleConfigs["no-hardcoded-env-urls"].SelectToken("allOf[0].properties.excludedhosts.default")!.Values().Select(v => v.ToString()).ToArray();
+
+            // From config
+            ConfigurationManager sut = new ConfigurationManager(new IOFileSystem());
+            RootConfiguration builtinConfig = sut.GetBuiltInConfiguration();
+            string[]? disallowedHostsInConfig = builtinConfig.Analyzers.GetValue<string[]?>("core.rules.no-hardcoded-env-urls.disallowedhosts", null);
+            disallowedHostsInConfig.Should().NotBeNull();
+            string[]? excludedHostsInConfig = builtinConfig.Analyzers.GetValue<string[]?>("core.rules.no-hardcoded-env-urls.excludedhosts", null);
+            excludedHostsInConfig.Should().NotBeNull();
+
+            disallowedHostsInSchema.Should().BeEquivalentTo(disallowedHostsInConfig, $"default of no-hardcoded-env-urls.disallowedHosts should be the same in {configPath} and {schemaPath}");
+            disallowedHostsInSchema.Should().BeInAscendingOrder($"default of no-hardcoded-env-urls.disallowedHosts should be in alphabetical order in {schemaPath}");
```

Comment:
>BeInAscendingOrder [](http://example.com/codeflow?start=45&length=18)

Nice :)

Reply:
What can I say, I guess I'm CDO (OCD in alphabetical order) :-)